transport/packet.go

```diff
@@ -42,16 +42,33 @@ func (e UDPEndpoint) Connect(ctx context.Context) (net.Conn, error) {
 	return e.Dialer.DialContext(ctx, "udp", e.Address)
 }
 
-// PacketListenerEndpoint is a [PacketEndpoint] that connects to the given address using the given [PacketListener].
-type PacketListenerEndpoint struct {
-	// The Dialer used to create the net.Conn on Connect(). Must be non nil.
+// PacketDialer provides a way to dial a destination and establish datagram connections.
+type PacketDialer interface {
+	// Dial connects to `raddr`.
+	// `raddr` has the form `host:port`, where `host` can be a domain name or IP address.
+	Dial(ctx context.Context, addr string) (net.Conn, error)
+}
+
+// UDPPacketDialer is a [PacketDialer] that uses the standard [net.Dialer] to dial.
+// It provides a convenient way to use a [net.Dialer] when you need a [PacketDialer].
+type UDPPacketDialer struct {
+	Dialer net.Dialer
+}
+
+var _ PacketDialer = (*UDPPacketDialer)(nil)
+
+// Dial implements [PacketDialer].Dial.
+func (d *UDPPacketDialer) Dial(ctx context.Context, addr string) (net.Conn, error) {
+	return d.Dialer.DialContext(ctx, "udp", addr)
+}
+
+// PacketListenerDialer is a [PacketDialer] that connects to the destination using the given [PacketListener].
+type PacketListenerDialer struct {
```

Comment:
Do we still need `PacketListener`? Can we just `UDPPacketDialer.Dial` and then `Read` from the returned `net.Conn`?